src/providers/s3Provider.ts

```diff
@@ -78,17 +87,19 @@ export class S3Provider implements Provider {
         directories.add(directory.Prefix!.substring(path.length));
       });
 
-      result.Contents?.forEach(object => {
-        if (object.Key!.endsWith('index.html')) {
-          hasIndexHtmlFile = true;
-        }
+      if (listFiles) {
+        result.Contents?.forEach(object => {
+          if (object.Key!.endsWith('index.html')) {
```

Comment:
index.html in a constant please :)

Also use a regex to check for index.html or index.htm